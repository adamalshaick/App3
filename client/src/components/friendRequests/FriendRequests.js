import React, { Component } from "react";
import RequestItem from "./RequestItem";
import PropTypes from "prop-types";

class FriendRequests extends Component {
  render() {
    const { profile, profiles } = this.props;
    return profile.friendRequests.map(request => (
      <RequestItem
        key={request._id}
        request={request}
        profile={profile}
        profiles={profiles}
      />
    ));
  }
}

FriendRequests.propTypes = {
  profile: PropTypes.object.isRequired
};

export default FriendRequests;
