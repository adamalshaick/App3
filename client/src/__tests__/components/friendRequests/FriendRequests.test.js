import React from "react";
import FriendRequests from "../../../components/friendRequests/FriendRequests";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

let wrapper;

const Props = {
  profile: {
    friendRequests: [{ _id: "id" }, { _id: "id" }]
  },

  profiles: {}
};

wrapper = shallow(<FriendRequests {...Props} />);

describe("render component", () => {
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
