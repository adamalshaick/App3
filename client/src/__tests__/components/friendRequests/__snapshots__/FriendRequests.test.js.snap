// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
Array [
  <Connect(RequestItem)
    key="id"
    profile={
      Object {
        "friendRequests": Array [
          Object {
            "_id": "id",
          },
          Object {
            "_id": "id",
          },
        ],
      }
    }
    profiles={Object {}}
    request={
      Object {
        "_id": "id",
      }
    }
  />,
  <Connect(RequestItem)
    key="id"
    profile={
      Object {
        "friendRequests": Array [
          Object {
            "_id": "id",
          },
          Object {
            "_id": "id",
          },
        ],
      }
    }
    profiles={Object {}}
    request={
      Object {
        "_id": "id",
      }
    }
  />,
]
`;
